test/it/uk/gov/hmrc/TestUserSpec.scala

```diff
@@ -0,0 +1,195 @@
+      val expectedIndividualCreated = TestIndividualCreatedResponse.from(individualFromMongo.copy(password = individualCreated.password))
+      individualCreated shouldBe expectedIndividualCreated
+      validatePassword(individualCreated.password, individualFromMongo.password) shouldBe true
+    }
+
+    scenario("Create an organisation") {
+
+      When("I request the creation of an organisation")
+      val createdResponse = Http(s"$serviceUrl/organisation").postForm.asString
+
+      Then("The response contains the details of the organisation created")
+      createdResponse.code shouldBe SC_CREATED
+      val organisationCreated = Json.parse(createdResponse.body).as[TestOrganisationCreatedResponse]
+
+      And("The organisation is stored in Mongo with hashed password")
+      val organisationFromMongo = result(mongoRepository.fetchByUsername(organisationCreated.username), timeout).get.asInstanceOf[TestOrganisation]
+      val expectedOrganisationCreated = TestOrganisationCreatedResponse.from(organisationFromMongo.copy(password = organisationCreated.password))
+      organisationCreated shouldBe expectedOrganisationCreated
+      validatePassword(organisationCreated.password, organisationFromMongo.password) shouldBe true
+    }
+  }
+
+  feature("Authenticate test user") {
+
+    def authenticate(usr: String, pwd: String): HttpResponse[String] = {
+      val payload = Json.parse(
+        s"""
+           |{
+           |  "username": "$usr",
+           |  "password" :"$pwd"
+           |}
+           |
+        """.stripMargin
+      ).toString
+
+      Http(s"$serviceUrl/authenticate")
+        .headers("Content-type" -> "application/json")
+        .postData(payload).asString
+    }
+
+    // considering the case of individual only, as for organisations it is similar
+    scenario("Valid credentials") {
+
+      When("An individual is created")
+      val individualCreatedResponse = Http(s"$serviceUrl/individual").postForm.asString
+      val individualCreated = Json.parse(individualCreatedResponse.body).as[TestIndividualCreatedResponse]
+      val individualFromMongo = result(mongoRepository.fetchByUsername(individualCreated.username), timeout).get.asInstanceOf[TestIndividual]
+
+      And("If I login with the individual's credentials")
+      val loginIndividualResponse = authenticate(individualCreated.username, individualCreated.password)
+
+      Then("The response contains the details of the individual created previously")
+      loginIndividualResponse.code shouldBe SC_OK
+      val individual = Json.parse(loginIndividualResponse.body).as[TestIndividualResponse]
+      val expectedIndividual = TestIndividualResponse.from(individualFromMongo)
+      individual shouldBe expectedIndividual
+
+      And("The response does not contain the password")
+      loginIndividualResponse.body.toLowerCase shouldNot include("password")
+
+      And("The response contains 'individual' as user type")
```

Comment:
more exactly this should be renamed "The response contains the details of the individual"